### PR TITLE
Misc fixes and improvements

### PR DESCRIPTION
**Pull request by tkreuzer**
This is a collection of misc fixes/improvements from the amd64-bringup branch.

### CODE REVIEW

File: boot/freeldr/freeldr/lib/peloader.c [hunk 1]
```diff
@@ -335,7 +336,7 @@ WinLdrLoadImage(IN PCHAR FileName,
 
         if (PhysicalBase == NULL)
         {
-            // Print(L"Failed to alloc pages for image %s\n", FileName);
+            ERR("Failed to alloc bytes for image %s\n", NtHeaders->OptionalHeader.SizeOfImage, FileName);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Forgot a format specifier for SizeOfImage

Comment (katahiromz) [reply boot/freeldr/freeldr/lib/peloader.c]: `%lu` or `0x%08X` for `DWORD`

File: boot/freeldr/freeldr/lib/peloader.c [hunk 2]
```diff
@@ -287,14 +287,15 @@ WinLdrLoadImage(IN PCHAR FileName,
     Status = ArcOpen(FileName, OpenReadOnly, &FileId);
     if (Status != ESUCCESS)
     {
-        // UiMessageBox("Can not open the file.");
+        ERR("ArcOpen(%s, %u, *) failed: 0x%lx\n", FileName, OpenReadOnly, Status);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Quote `%s` for `FileName`.

File: boot/freeldr/freeldr/lib/peloader.c [hunk 3]
```diff
@@ -304,7 +305,7 @@ WinLdrLoadImage(IN PCHAR FileName,
     NtHeaders = RtlImageNtHeader(HeadersBuffer);
     if (!NtHeaders)
     {
-        // Print(L"Error - no NT header found in %s\n", FileName);
+        ERR("No NT header found in %s\n", FileName);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Quote `%s` for `FileName`.

File: boot/freeldr/freeldr/lib/peloader.c [hunk 4]
```diff
@@ -313,7 +314,7 @@ WinLdrLoadImage(IN PCHAR FileName,
     /* Ensure this is executable image */
     if (((NtHeaders->FileHeader.Characteristics & IMAGE_FILE_EXECUTABLE_IMAGE) == 0))
     {
-        // Print(L"Not an executable image %s\n", FileName);
+        ERR("Not an executable image %s\n", FileName);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Quote `%s` for `FileName`.

File: boot/freeldr/freeldr/lib/peloader.c [hunk 5]
```diff
@@ -352,6 +353,7 @@ WinLdrLoadImage(IN PCHAR FileName,
     Status = ArcSeek(FileId, &Position, SeekAbsolute);
     if (Status != ESUCCESS)
     {
+        ERR("ArcSeek failed. Status=0x%lx, file %s\n", Status, FileName);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Quote `%s` for `FileName`.

File: boot/freeldr/freeldr/lib/peloader.c [hunk 6]
```diff
@@ -360,7 +362,7 @@ WinLdrLoadImage(IN PCHAR FileName,
     Status = ArcRead(FileId, PhysicalBase, NtHeaders->OptionalHeader.SizeOfHeaders, &BytesRead);
     if (Status != ESUCCESS)
     {
-        // Print(L"Error reading headers %s\n", FileName);
+        ERR("ArcRead failed. Status=0x%lx, file %s\n", Status, FileName);
```
Comment (katahiromz) [inline boot/freeldr/freeldr/lib/peloader.c]: Quote `%s` for `FileName`.

File: ntoskrnl/io/iomgr/device.c [hunk 7]
```diff
@@ -1084,7 +1084,8 @@ IoCreateDevice(IN PDRIVER_OBJECT DriverObject,
     if (DeviceName) ObjectAttributes.Attributes |= OBJ_PERMANENT;
 
     /* Align the Extension Size to 8-bytes */
```
Comment (katahiromz) [inline ntoskrnl/io/iomgr/device.c]: Fix this comment.

File: ntoskrnl/ke/i386/ctxswitch.S [hunk 8]
```diff
@@ -61,12 +61,12 @@ PUBLIC @KiSwapContextInternal@0
  * BOOLEAN FASTCALL
  * KiSwapContext(PKTHREAD CurrentThread, PKTHREAD TargetThread);
  *
+ * \param WaitIrql<cl>
```
Comment (HBelusca) [inline ntoskrnl/ke/i386/ctxswitch.S]: "<cl>" ??

File: ntoskrnl/ke/i386/ctxswitch.S [hunk 9]
```diff
@@ -61,12 +61,12 @@ PUBLIC @KiSwapContextInternal@0
  * BOOLEAN FASTCALL
  * KiSwapContext(PKTHREAD CurrentThread, PKTHREAD TargetThread);
```
Comment (HBelusca) [inline ntoskrnl/ke/i386/ctxswitch.S]: This list of parameters do not correspond to the modified explanations given below.

File: ntoskrnl/ke/i386/ctxswitch.S (hunk 8 above)
Comment (tkreuzer) [reply ntoskrnl/ke/i386/ctxswitch.S]: !!

File: boot/freeldr/freeldr/lib/peloader.c [hunk 10]
```diff
@@ -287,14 +287,15 @@ WinLdrLoadImage(IN PCHAR FileName,
     Status = ArcOpen(FileName, OpenReadOnly, &FileId);
     if (Status != ESUCCESS)
     {
-        // UiMessageBox("Can not open the file.");
```
Comment (SergeGautherie) [inline boot/freeldr/freeldr/lib/peloader.c]: NB: Not sure why `UiMessageBox()` is unwanted here, as there seems to be one in the other cases?

File: ntoskrnl/ke/i386/ctxswitch.S (hunk 8 above)
Comment (HBelusca) [reply ntoskrnl/ke/i386/ctxswitch.S]: hm, now that I think again about it, isn't it the register name?

File: ntoskrnl/ex/uuid.c [hunk 11]
```diff
@@ -339,27 +339,9 @@ VOID
...
-        {
-            break;
-        }
-
-        PrevLuid.QuadPart = CompLuid;
-    }
-
-    LocallyUniqueId->LowPart = PrevLuid.LowPart;
-    LocallyUniqueId->HighPart = PrevLuid.HighPart;
+    *(LONG64*)LocallyUniqueId = InterlockedExchangeAdd64((PLONG64)&ExpLuid,
```
Comment (ThFabba) [inline ntoskrnl/ex/uuid.c]: Does the RHS work without the cast if you use `.QuadPart`?

File: dll/win32/shlwapi/ordinal.c [hunk 12]
```diff
@@ -1246,10 +1246,10 @@ BOOL WINAPI SHIsSameObject(IUnknown* lpInt1, IUnknown* lpInt2)
   if (lpInt1 == lpInt2)
     return TRUE;
 
-  if (IUnknown_QueryInterface(lpInt1, &IID_IUnknown, (void**)&lpUnknown1) != S_OK)
+  if (FAILED(IUnknown_QueryInterface(lpInt1, &IID_IUnknown, (PVOID*)&lpUnknown1)))
```
Comment (ThFabba) [inline dll/win32/shlwapi/ordinal.c]: This is Wine code, this doesn't seem worth a diff.

File: dll/directx/ksuser/ksuser.c [hunk 13]
```diff
@@ -195,7 +195,7 @@ KsCreatePin(HANDLE FilterHandle,
             PHANDLE  ConnectionHandle)
 {
     ULONG BufferSize = sizeof(KSPIN_CONNECT);
-    PKSDATAFORMAT DataFormat = ((PKSDATAFORMAT) ((ULONG_PTR)Connect + sizeof(KSPIN_CONNECT)));
+    PKSDATAFORMAT DataFormat = (PKSDATAFORMAT)(Connect + 1);
```
Comment (ThFabba) [inline dll/directx/ksuser/ksuser.c]: I prefer this way of writing it too, but does this really make any difference?

File: boot/freeldr/freeldr/lib/peloader.c (hunk 10 above)
Comment (SergeGautherie) [reply boot/freeldr/freeldr/lib/peloader.c]: It seems to have been commented out on r46017.

File: dll/directx/ksuser/ksuser.c (hunk 13 above)
Comment (tkreuzer) [reply dll/directx/ksuser/ksuser.c]: It makes a difference for my eyes (bleeding vs not bleeding).

File: ntoskrnl/ex/uuid.c (hunk 11 above)
Comment (tkreuzer) [reply ntoskrnl/ex/uuid.c]:
```
typedef struct _LUID {
    ULONG LowPart;
    LONG HighPart;
} LUID, *PLUID;
```

**Pull request merged**